milli/src/update/index_documents/extract/extract_fid_docid_facet_values.rs

```diff
@@ -79,7 +93,8 @@ pub fn extract_fid_docid_facet_values<R: io::Read + io::Seek>(
 
     Ok((
         sorter_into_reader(fid_docid_facet_numbers_sorter, indexer.clone())?,
-        sorter_into_reader(fid_docid_facet_strings_sorter, indexer)?,
+        sorter_into_reader(fid_docid_facet_strings_sorter, indexer.clone())?,
+        facet_exists_docids,
```

Comment:
I'm a bit concerned by this btreemap return, I understand that its size is small, but, we are in a multi-threaded process where threads share and maximize the usage of memory. I feel like the memory used by this btreemap "leaks" out of the process.
I suggest finding a way to write the data on the disk before quitting the function.
I don't know if I'm clear, so don't hesitate to ask me to re-phrase it.

Reply:
You're right. I did think it was too small to matter at the beginning, but if we have 10 million documents and 100 faceted fields, it could get quite large! I will rework it :)